web3.js/src/transaction.ts

```diff
@@ -221,11 +269,46 @@ export class Transaction {
    */
   _json?: TransactionJSON;
 
+  // Construct a transaction with a blockhash and lastValidBlockHeight
+  constructor(opts?: TransactionBlockhashCtor);
+
+  // Construct a transaction with a nonce
+  constructor(opts?: TransactionNonceCtor);
+
+  /**
+   * @deprecated `recentBlockhash` has been deprecated.
+   * Please pass `latestBlockhash: {blockHash: BlockHash, lastValidBlockHeight: number}` instead.
+   * This will be removed in a future version of @solana/web3.js.
+   */
+  constructor(opts?: TransactionCtorFields);
+
   /**
    * Construct an empty Transaction
    */
-  constructor(opts?: TransactionCtorFields) {
-    opts && Object.assign(this, opts);
+  constructor(
+    opts?:
+      | TransactionBlockhashCtor
+      | TransactionNonceCtor
+      | TransactionCtorFields,
```

Comment:
Still think this type should have a `_DEPRECATED` suffix, to make it super clear.

Reply:
Okay, I'll add that back in. :) 

Does the rest of it look okay to you?